app/models/host/discovered.rb

```diff
@@ -108,6 +109,16 @@ def populate_fields_from_facts(facts = self.facts_hash, ignored_type)
     end
     self.discovery_attribute_set = DiscoveryAttributeSet.where(:host_id => id).first_or_create
     self.discovery_attribute_set.update_attributes(import_from_facts)
+    # lock the host into discovery via PXE, if feature is enabled in settings
+    if Setting::Discovered.discovery_lock? and self.subnet.tftp?
+      begin
+        Rails.logger.info "Locking discovered host #{self.mac} in subnet #{subnet}"
+        template = unattended_render(::ProvisioningTemplate.find_by_name(Setting['discovery_lock_template']).template)
```

Comment:
No need to explicitly call `template` method, the renderer will call it for you I think.